controllers/hivelocitycluster_controller.go

```diff
@@ -192,16 +192,15 @@ func (r *HivelocityClusterReconciler) reconcileNormal(ctx context.Context, clust
 		if err != nil {
 			return ctrl.Result{}, fmt.Errorf("failed to get HivelocityMachineTemplate %q: %w", name, err)
 		}
-		machineType := hmt.Spec.Template.Spec.Type
-		if machineType == "" {
-			return ctrl.Result{}, fmt.Errorf("Spec.Template.Spec.Type of HivelocityMachineTemplate %q is empty", name)
+		deviceSelector := hmt.Spec.Template.Spec.DeviceSelector
```

Comment:
we should be able to cope with an empty selector

Reply:
in the next line i check similar things right?

Reply:
yes you can remove all of it